main.go

```diff
@@ -0,0 +1,196 @@
+
+	configv1 "github.com/openshift/api/config/v1"
+	opmetrics "github.com/openshift/operator-custom-metrics/pkg/metrics"
+	splunkforwarderv1alpha1 "github.com/openshift/splunk-forwarder-operator/api/v1alpha1"
+	"github.com/openshift/splunk-forwarder-operator/config"
+	"github.com/openshift/splunk-forwarder-operator/controllers/secret"
+	"github.com/openshift/splunk-forwarder-operator/controllers/splunkforwarder"
+	"github.com/openshift/splunk-forwarder-operator/version"
+	"github.com/operator-framework/operator-lib/leader"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	monclientv1 "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned/typed/monitoring/v1"
+	//+kubebuilder:scaffold:imports
+)
+
+const (
+	// Environment variable to determine operator run mode
+	ForceRunModeEnv = "OSDK_FORCE_RUN_MODE"
+	// Flags that the operator is running locally
+	LocalRunMode = "local"
+)
+
+var (
+	metricsPort int32 = 8383
+	scheme            = apiruntime.NewScheme()
+	setupLog          = ctrl.Log.WithName("setup")
+)
+var log = logf.Log.WithName("cmd")
+
+func printVersion() {
+	log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	log.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+	log.Info(fmt.Sprintf("Version of operator-sdk: %v", version.SDKVersion))
+}
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(splunkforwarderv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(configv1.Install(scheme))
+	utilruntime.Must(monitoringv1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":"+fmt.Sprintf("%d", metricsPort), "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	printVersion()
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "f48c54da.managed.openshift.io",
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	ctx := context.TODO()
+
+	// Become the leader before proceeding
+	// This doesn't work locally, so only perform it when running on-cluster
+	if strings.ToLower(os.Getenv(ForceRunModeEnv)) != LocalRunMode {
```

Comment:
This seems strange, couldn't we control this behavior with the `leader-elect`/`enableLeaderElection` flag/bool when creating the new manager?

Reply:
The `leader-elect` flag, when passed to the controller manager, will set it up to perform leader-with-lease election. Whereas using `Become()` will use leader-for-life election.

Either option is validly supported in the SDK, and neither is the "right" answer - SDK 1.x defaults to lease, but leaves the decision ultimately up to the operator author. Currently, SFO performs leader-for-life, so as part of SDE-1972 we've been trying to avoid subtly changing operator behaviour unless necessary - hence maintaining leader-for-life here.

Ultimately, given the operator runs 1 replica, I suspect whichever method is chosen is largely a non-issue. But I'll leave that decision up to yourselves as the operator custodians. :)  Happy to switch to a leader-with-lease approach if you'd prefer that, and remove this block - just let me know if that suits.

Reply:
I will note that, having just attempted to test the switch to leader-by-lease, I can observe that it would also require the operator serviceaccount's RBAC to require additional permissions, because currently it will fail with:

```
E0815 22:28:12.477163       1 leaderelection.go:330] error retrieving resource lock openshift-splunk-forwarder-operator/f48c54da.managed.openshift.io: leases.coordination.k8s.io "f48c54da.managed.openshift.io" is forbidden: User "system:serviceaccount:openshift-splunk-forwarder-operator:splunk-forwarder-operator" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "openshift-splunk-forwarder-operator"
```

Reply:
Thanks for the information! Honestly I am not the most familiar with SFO haha, though reviewing the documentation I agree with your assessment that this is likely a non-issue either way